interfaces/AW-App/src/app/validation-components/upload-data/upload-data.component.ts

```diff
@@ -93,18 +108,13 @@ export class UploadDataComponent implements ValidationComponent {
     }
   }
 
-  public async removeCredentialByDid(did: string): Promise<void> {
-    const currentCredentials = await this.storage.get(
-      this.ionicStorageTypes.credentials,
-    );
-    currentCredentials.splice(
-      currentCredentials.findIndex((item) => item.did === did),
-      1,
-    );
-    await this.storage.set(
-      this.ionicStorageTypes.credentials,
-      currentCredentials,
-    );
+  public async removeLocalStorageData(
+    did: string,
+    ionicStorageType: IonicStorageTypes,
+  ): Promise<void> {
+    let data = await this.storage.get(this.ionicStorageTypes[ionicStorageType]);
+    data = data.filter((item) => item.did !== did);
+    await this.storage.set(this.ionicStorageTypes[ionicStorageType], data);
```

Comment:
The value of the parameter `ionicStorageType` is already the correct value, I think.

```suggestion
  public async removeLocalStorageData(
    did: string,
    type: IonicStorageTypes,
  ): Promise<void> {
    let data = await this.storage.get(type);
    data = data.filter((item) => item.did !== did);
    await this.storage.set(type, data);
```

Should also work. (Haven't tested it though..)